### PR TITLE
[router]: add sitemap option to disable sitemap generation

### PR DESCRIPTION
**Pull request by marklawlor**
# Why

Not every website requires a sitemap (or they may generate their own custom one). This option allows users to disable this setting.

# How

Add a new config plugin setting `sitemap`. When set to `false` the sitemap will, not be generated.

I considered making this an option to disable the other generated files, but they don't make sense to disable.

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-router/src/getRoutesCore.ts [hunk 1]
```diff
@@ -287,7 +288,7 @@ function getDirectoryTree(contextModule: RequireContext, options: Options) {
 
   // Only include the sitemap if there are routes.
   if (!options.skipGenerated) {
-    if (hasRoutes) {
+    if (hasRoutes && options.sitemap !== false) {
```
Comment (EvanBacon) [inline packages/expo-router/src/getRoutesCore.ts:291]: run this and verify it works, the config plugin isn't populating this method.

**Comment by EvanBacon** [comment]
make sure to update the docs https://docs.expo.dev/router/reference/sitemap/

File: docs/pages/router/reference/sitemap.mdx [hunk 2]
```diff
@@ -27,7 +27,24 @@ You can also search for links directly in a browser like Safari or Chrome to tes
 
 Expo Router currently injects a **/\_sitemap** automatically that provides a list of all routes in the app. This is useful for debugging.
 
-To remove it, add a **app/\_sitemap.tsx** and return a null component:
+In SDK 52+, the sitemap can be removed by adding `sitemap: false` to your `app.json`
```
Comment (amandeepmittal) [inline docs/pages/router/reference/sitemap.mdx:30]:
```suggestion
In SDK 52 and above, the sitemap can be removed by adding `sitemap: false` to the `expo-router` config plugin in the app config:
```

File: docs/pages/router/reference/sitemap.mdx [hunk 3]
```diff
@@ -27,7 +27,24 @@ You can also search for links directly in a browser like Safari or Chrome to tes
...
+        /* @info Disable the sitemap generation. */
+        "sitemap": false
+        /* @end */
+      }
+    ]
+  ]
+}
+```
+
+In SDK <52, the sitemap can be removed by creating an empty `_sitemap` file.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/sitemap.mdx:47]:
```suggestion
In SDK 51 and below, the sitemap can be removed by creating an empty `_sitemap` file.
```

File: docs/pages/router/reference/sitemap.mdx [hunk 4]
```diff
@@ -27,7 +27,24 @@ You can also search for links directly in a browser like Safari or Chrome to tes
...
+        /* @info Disable the sitemap generation. */
+        "sitemap": false
+        /* @end */
+      }
+    ]
+  ]
+}
+```
+
+In SDK 51 and below, the sitemap can be removed by creating an empty `_sitemap` file.
```
Comment (amandeepmittal) [inline docs/pages/router/reference/sitemap.mdx:47]:
```suggestion
In SDK 51 and below, the sitemap can be removed by creating an empty **_sitemap.tsx** file inside **app** directory.
```

Let's make file names bold as per our writing style guide and also mention that create this file inside app directory.

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**